src/compiler/checker.ts

```diff
@@ -7174,8 +7175,12 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     context.tracker.reportNonSerializableProperty(symbolToString(propertySymbol));
                 }
             }
-            context.enclosingDeclaration = propertySymbol.valueDeclaration || propertySymbol.declarations?.[0] || saveEnclosingDeclaration;
+            const propertyDeclaration = propertySymbol.valueDeclaration || propertySymbol.declarations?.[0];
+            context.enclosingDeclaration = propertyDeclaration || saveEnclosingDeclaration;
             const propertyName = getPropertyNameNodeForSymbol(propertySymbol, context);
+            if (propertyDeclaration && (isPropertyAssignment(propertyDeclaration) || isShorthandPropertyAssignment(propertyDeclaration) || isMethodDeclaration(propertyDeclaration) || isMethodSignature(propertyDeclaration) || isPropertySignature(propertyDeclaration) || isPropertyDeclaration(propertyDeclaration) || isGetOrSetAccessorDeclaration(propertyDeclaration))) {
+                setSourceMapRange(propertyName, propertyDeclaration.name);

```

Comment:
For declaration emit nodes, just use `setTextRange` rather than `setSourceMapRange` - there's a helper in scope that takes care of a bunch of sanity checking for you that's absent here. But yeah, you should also be able to move this into `getPropertyNameNodeForSymbol` so it happens for all callers.

Reply:
Using `setTextRange` doesn't cut it and the results are as on the `main` branch. Am I using it wrong or missing something?

I have moved the fix to `getPropertyNameNodeForSymbol` though

Reply:
Yeah, it doesn't seem to help. I don't understand why this isn't working:

```ts
propertyNameNode = setTextRange(context, propertyNameNode, declaration.name);
```

So I would expect the original node stuff to "just work". `setTextRange` doesn't call `setSourceMapRange` at all, though, so...

Reply:
If you stick `setSourceMapRange(range, location);` into `setTextRange` after `setOriginalNode`, that actually also changes a whole bunch of other stuff kinda for the better? But then breaks some other JSDoc related code (which I think was another recent bugfix that may conflict?) Dunno if that's a good idea or not, I don't have a good mental model of this.

Reply:
(maybe #59675?)